src/sagemaker/estimator.py

```diff
@@ -614,16 +614,24 @@ def __init__(
         self.output_kms_key = resolve_value_from_config(
             output_kms_key, TRAINING_JOB_KMS_KEY_ID_PATH, sagemaker_session=self.sagemaker_session
         )
+        use_volume_kms_config: bool = False
         if instance_type is None or isinstance(instance_type, str):
             instance_type_for_volume_kms = instance_type
         elif isinstance(instance_type, ParameterString):
             instance_type_for_volume_kms = instance_type.default_value
         else:
-            raise ValueError(f"Bad value for instance type: '{instance_type}'")
+            # Always attach a KMS key if an instance type that isn't a string or parameter
```

Comment:
I guess parameter here means PipelineParameter. It should not be excluded from the PipelineVariable. See the comment above for context.